clients/unshaded/src/main/java/tachyon/client/BlockOutStream.java

```diff
@@ -61,7 +61,9 @@ public static BlockOutStream get(TachyonFile tachyonFile, WriteType opType, int
 
     if (tachyonFile.mTachyonFS.hasLocalWorker()
         && tachyonConf.getBoolean(Constants.USER_ENABLE_LOCAL_WRITE,
-            Constants.DEFAULT_USER_ENABLE_LOCAL_WRITE)) {
+            Constants.DEFAULT_USER_ENABLE_LOCAL_WRITE)
+            && tachyonConf.get(Constants.WORKER_BALANCER_STRATEGY_TYPE, "DEFAULT")
+                   .equalsIgnoreCase("DEFAULT")) {
```

Comment:
If I don't had this condition, if there is a local worker we will always choose this local worker even if a remote one has more free space. Or is there sth I ve misunderstood ?